tasks/install.yml

```diff
@@ -1,38 +1,30 @@
 ---
-- name: 'Download binary from https://releases.mattermost.com'
-  get_url: 
-    url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz'
-    dest: "{{ global_cache_dir | mandatory }}/"
-  when: not mattermost_enterprise
-  become: false
-  delegate_to: localhost
 
-- name: 'Download enterprise version binary from https://releases.mattermost.com'
-  get_url: 
-    url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz'
-    dest: "{{ global_cache_dir | mandatory }}/"
-  when: mattermost_enterprise
-  become: false
-  delegate_to: localhost
+- name: Check if a previous update wasn't deployed
+  stat:
+    path: /opt/mattermost-updated/mattermost
+  register: mattermost_updated_stat
 
-- name: Unpack Mattermost archive
-  unarchive: 
-    src: "{{ global_cache_dir }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz"
-    dest: /opt/
-  args:
-    creates: /opt/mattermost/bin/platform
-  when: not mattermost_enterprise
+- import_tasks: deploy.yml
+  when: mattermost_updated_stat.stat.exists
 
-- name: Unpack Mattermost enterprise archive
-  unarchive: 
-    src: "{{ global_cache_dir }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz"
-    dest: /opt/
-  args:
-    creates: /opt/mattermost/bin/platform
-  when: mattermost_enterprise
+- name: Check if a previous mattermost version exists
+  stat:
+    path: /opt/mattermost/version
+  register: previous_mattermost_version_stat
 
-- name: Create user to run Mattermost service
-  user: 
-    name: "{{ mattermost_user }}"
-    system: yes
-    createhome: no 
+- name: Get the previous mattermost version
+  slurp:
+    src: /opt/mattermost/version
+  register: previous_mattermost_version
+  when: previous_mattermost_version_stat.stat.exists
+
+- name: Set that mattermost needs to be updated
+  set_fact:
+    mattermost_needs_update:
```

Comment:
You could also just set the variable's content to the condition. This way, you just need to check the variable and not whether it's defined

Reply:
This is true in the context of `install.yml`, but I wanted `update.yml` to be self-contained. No matter of the previous state, running `update.yml` works and places the updated version inside `/opt/mattermost-updated/mattermost`.